### PR TITLE
Remove buggy BarrierStat

### PR DESCRIPTION
**Pull request by reyoung**
The implementation of BarrierStat is buggy, and it is not necessary
for Paddle to diagnose which node in cluster is slow.

### CODE REVIEW

File: paddle/pserver/ParameterServer2.cpp [hunk 1]
```diff
@@ -369,50 +365,7 @@ void ParameterServer2::addGradient(const SendParameterRequest& request,
...
-        isSparseServer_ ? "_sparseUpdater" : "_denseUpdater");
-  }
-
   {
-    /// approximately pure network overhead
-    REGISTER_TIMER_DYNAMIC_SET(
-        "pushRecv", timeToMicroSecond(*handleRequestBegin_), -1, *statSet_);
-  }
-
-#ifndef PADDLE_DISABLE_TIMER
```
Comment (typhoonzero) [inline paddle/pserver/ParameterServer2.cpp]: Should also remove the cmake flag `WITH_TIMER`

File: paddle/pserver/ParameterServer2.cpp [hunk 2]
```diff
@@ -369,50 +365,7 @@ void ParameterServer2::addGradient(const SendParameterRequest& request,
...
-  if (!numPassFinishClients_) {
-    REGISTER_BARRIER_TIMER_SERVER(
-        *statSet_,
-        "addGradBegin",
-        FLAGS_num_gradient_servers,
-        request.trainer_id(),
-        isSparseServer_ ? "_sparseUpdater" : "_denseUpdater");
-  }
-
-  {
```
Comment (typhoonzero) [inline paddle/pserver/ParameterServer2.cpp]: Should not remove this `{`

File: paddle/pserver/ParameterServer2.cpp (hunk 1 above)
Comment (reyoung) [inline paddle/pserver/ParameterServer2.cpp]: WITH_TIMER is not totally removed. It is used to calculate execution time for each layer and whole neural network in trainer side.

This PR only removes the `BarrierStats`, that used in PServer

File: paddle/pserver/ParameterServer2.cpp (hunk 2 above)
Comment (reyoung) [inline paddle/pserver/ParameterServer2.cpp]: This is a `diff` display error. That `{` is in line `384/368`

File: paddle/pserver/ParameterServer2.cpp (hunk 1 above)
Comment (typhoonzero) [inline paddle/pserver/ParameterServer2.cpp]: I see.

File: paddle/pserver/ParameterServer2.cpp (hunk 2 above)
Comment (typhoonzero) [inline paddle/pserver/ParameterServer2.cpp]: Got it

**Pull request merged**